### DISCUSSION TITLE
Migrate Workflow Definition After Task Rename

### DISCUSSION THREAD

**Discussion by davidpuplava** [OrchardCMS/OrchardCore#15928] [Q&A] [answered]
What is the recommended way to migrate/update an existing workflow definition when an existing workflow task name changes? 

For example, suppose the UpdateTwitterStatusTask type needs to be renamed to UpdateXStatusTask, then my existing workflow content item that uses the UpdateTwitterStatusTask will need to be updated to reference UpdateXStatusTask.

Is a DataMigration the right way? I see the documentation talks about updating content types, but can content items themselves be updated in a DataMigration?

**Comment by Piedone** [comment, marked as answer]
Migrations can be used for this, yes, since this should be done automatically. You'll need to loop through all Workflow Types and update them (perhaps current running Workflow instances too). If the expected count of items here is low (for Workflow Types it is, since a large site may have like 50 workflows, but that's still a small amount) then there's nothing special to be done. If you also need to update Workflow instances, then that's harder, because there can be tens of thousands.

**Reply by davidpuplava** [reply]
Thank you, @Piedone! I see something similar was done [here](https://github.com/OrchardCMS/OrchardCore/pull/11052/files#diff-74a3ba1fb26c512218a56465ffdfdc50b501eebab27ba6b14ae96161968e8214R152) for the Lucene search module rename.

**Reply by Piedone** [reply]
Indeed! No problem.
